src/Sparrow/AsyncManualResetEvent.cs

```diff
@@ -67,19 +62,6 @@ public Task<bool> WaitAsync()
             [Pure]
             public async Task<bool> WaitAsync(TimeSpan timeout)
             {
-                Debug.Assert(timeout != TimeSpan.MaxValue);
```

Comment:
OK. So not it does not throw as we will change it to unit.MaxValue - 1 when we create the timer.